docs/src/main/asciidoc/extension-writing-dev-service.adoc

```diff
@@ -5,118 +5,149 @@ https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 ////
 [id="extension-writing-dev-service"]
 = Writing a Dev Service
+
 include::_attributes.adoc[]
 :categories: writing-extensions
 :diataxis-type: howto
 :topics: extensions
-////
-////
 
+Learn how to develop a xref:dev-services.adoc[Dev Service] for your extension in order to replace an external service in development mode.
 
 == Prerequisites
 
-- You should already have an xref:building-my-first-extension.adoc[extension structure] in place
-- You should have a containerised version of your external service (not all Dev Services rely on containers, but most do)
+:prerequisites-time: 15 minutes
+:prerequisites-docker:
+:prerequisites-no-cli:
+:prerequisites-no-graalvm:
+include::{includes}/prerequisites.adoc[]
+* An xref:building-my-first-extension.adoc[extension structure] in place
+* A containerised version of your external service (not all Dev Services rely on containers, but most do)
 
 == Creating a Dev Service
 
-If your extension provides APIs for connecting to an external service, it's a good idea to provide a xref:dev-services.adoc[Dev Service] implementation.
+If your extension provides APIs for connecting to an external service, it's a good idea to provide a dev service implementation.
+
+First, you must add the following dependency to your build file :
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-devservices-deployment</artifactId>
+</dependency>
+----
 
-To create a Dev Service, add a new build step into the extension processor class that returns a `DevServicesResultBuildItem`.
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-devservices-deployment")
+----
+
+Then, add a new build step into the extension processor class that returns a `DevServicesResultBuildItem`.
 Here, the https://hub.docker.com/_/hello-world[`hello-world`] image is used, but you should set up the right image for your service.
 
-[source%nowrap,java]
+[source,java]
 ----
-    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = GlobalDevServicesConfig.Enabled.class)
-    public DevServicesResultBuildItem createContainer() {
-        DockerImageName dockerImageName = DockerImageName.parse("hello-world");
-        GenericContainer container = new GenericContainer<>(dockerImageName)
-                                         .withExposedPorts(SERVICE_PORT, OTHER_SERVICE_PORT)
-                                         .waitingFor(Wait.forLogMessage(".*" + "Started" + ".*", 1))
-                                         .withReuse(true);
```

Comment:
I think it was done this way to be clear it was a method of a class.

I'm not sure what's best really so I'm not asking you to change it, just stating the why.